### ISSUE TITLE
RegisterAssembly additional feature

### ISSUE THREAD

**Issue by Centurys** [seesharper/LightInject#392]
Hi,

we are using "RegisterAssembly" method to dynamically register services from assembly with defined lifetime and some search pattern. It's work for most situations, but there are need for additional feature to specify dynamically found service name, when registering service.
Would you consider extending/updating "RegisterAssembly" method with adding maybe additional parameter to enable return/set service name by found type?

**Comment by Centurys** [comment]
Any opinion on this @seesharper ?

**Comment by seesharper** [comment]
One thing is for sure and that is that we need to do it without introducing breaking changes. Maybe we can somehow create an abstraction for the code that creates the service name.

**Comment by seesharper** [comment]
Maybe something like 
```
public interface IServiceNameProvider
{
   string GetServiceName(Type serviceType, Type implementingType)
}
```

**Comment by Centurys** [comment]
I was thinking more of another overload on RegisterAssembly method. Could make another parameter like "Func<Type, Type, bool> shouldRegister", but for providing name e.g.  "Func<Type, Type, string> nameProvider". And only if new overload is used service name is provided, but other method overload do not have any changes.
With another interface there will be more coding, but maybe it could be used more broadly in library ...

In the end you have to choose implementation details :)

**Comment by seesharper** [comment]
@Centurys Fixed in LightInject 5.1.2 👍

**Comment by Centurys** [comment]
Much appreciated :)
